packages/firebase_dynamic_links/android/src/main/java/io/flutter/plugins/firebasedynamiclinks/FirebaseDynamicLinksPlugin.java

```diff
@@ -114,6 +117,32 @@ public void onMethodCall(MethodCall call, Result result) {
     return dynamicLink;
   }
 
+  private void handleGetDynamicLink(final Result result, Uri uri) {
+    FirebaseDynamicLinks.getInstance()
+        .getDynamicLink(uri)
```

Comment:
This call is very similar to initialDynamicLink call.

I suggest moving that to a private method and calling it both from this method and the initial dynamic link method. It will also get rid of some of the inconsistencies. For instance, do the listeners really need activity? If not, we should remove it from both. We should return the pendingDynamicLinkData from both.

Reply:
Updated. Thanks.